tests/main/snap-snap-uri-handler/task.yaml

```diff
@@ -0,0 +1,51 @@
+summary: Ensure the "snap snap-uri-handler" command works
+
+# We don't run on ubuntu-core, since we are testing the behaviour of
+# xdg-open on the host system.
+systems: [-ubuntu-core-*]
+
+# xdg-open doesn't seem to check desktop files if it doesn't think it
+# has a display.
+environment:
+    DISPLAY: :placeholder
+
+prepare: |
+    #shellcheck source=tests/lib/pkgdb.sh
+    . "$TESTSLIB/pkgdb.sh"
+    distro_install_package xdg-utils
+    touch /usr/bin/zenity
+
+restore: |
+    #shellcheck source=tests/lib/pkgdb.sh
+    . "$TESTSLIB/pkgdb.sh"
+    umount -f /usr/bin/zenity || :
+    distro_purge_package xdg-utils
+    distro_auto_remove_packages
+
+execute: |
+    #shellcheck source=tests/lib/pkgdb.sh
+    . "$TESTSLIB/pkgdb.sh"
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+
+    echo "URI Handler fails if gnome-software is not installed and user refuses to install it"
+    mount --bind /bin/false /usr/bin/zenity
+    if snap snap-uri-handler snap://package 2>errors.log; then
```

Comment:
Can't that be just something like:
```
snap handle-link snap://package 2>&1 | MATCH "GNOME Software required"
```
?

Reply:
We want to both check the exit code of the command invocation and that the expected message was printed to stderr (not stdout).